### ISSUE TITLE
NullPointerException (NPE) in ArrayMemberValue in 3.18.1-GA

### ISSUE THREAD

**Issue by Raymond-Naseef** [jboss-javassist/javassist#80]
This happens whenever annotation contains member that is empty.  I can create test code to display this, but think the issue is probably simple, with solution be to check if field values is null.  This can be reproduced with javax.persistence.Table annotation.

java.lang.NullPointerException: null
	at javassist.bytecode.annotation.ArrayMemberValue.write(ArrayMemberValue.java:133) ~[javassist-3.18.1-GA.jar:na]
	at javassist.bytecode.annotation.Annotation.write(Annotation.java:318) ~[javassist-3.18.1-GA.jar:na]
	at javassist.bytecode.AnnotationsAttribute.setAnnotations(AnnotationsAttribute.java:248) ~[javassist-3.18.1-GA.jar:na]

Here is code causing the error, with workaround commented out that will avoid NPE:
```
			Annotation[] annotations = new Annotation[2];
			AnnotationsAttribute attrib =
					new AnnotationsAttribute(constPool, AnnotationsAttribute.visibleTag);
			Annotation annotation = new Annotation(constPool, classPool.get(Entity.class.getName()));
			annotations[0] = annotation;
			// Add @Table(name="",schema="") to class
			annotation = new Annotation(constPool, classPool.get(Table.class.getName()));
			annotation.addMemberValue("name", new StringMemberValue(dataTableName, constPool));
			annotation.addMemberValue("schema", new StringMemberValue(dataSchema, constPool));
			// Add empty array to remove NPE at javassist.bytecode.annotation.ArrayMemberValue.write(ArrayMemberValue.java:133)
			// *** THIS FIX (workaround) NEEDS UNCOMMENTING THESE 4 LINES
//			ArrayMemberValue blankMemberValueArray = new ArrayMemberValue(new AnnotationMemberValue(constPool), constPool);
//			blankMemberValueArray.setValue(new MemberValue[0]);
//			annotation.addMemberValue("uniqueConstraints", blankMemberValueArray);
//			annotation.addMemberValue("indexes", blankMemberValueArray);
			annotations[1] = annotation;
			attrib.setAnnotations(annotations);
			ctFile.addAttribute(attrib);

```

**Comment by chibash** [comment]
I couldn't reproduce this bug.  Maybe the latest version has already fixed this problem.

**Comment by Raymond-Naseef** [comment]
Sorry, did not look into that: will find out and let you know if the issue is solved.

**Comment by Raymond-Naseef** [comment]
I still see the problem with latest Javassist 3.20.0-GA - will provide entire running example of the problem.

**Comment by Raymond-Naseef** [comment]
I created project showing the bug - too detailed.  Found your test/Test.java with my workaround.  I commented it out in fork of github: you should be able to see this easily by commenting out the fix as you can see in my code above.  The bug is ultra-simple: checking field 'values' that was never assigned.

**Comment by Raymond-Naseef** [comment]
Checked the Test.java file: that will not show the bug because @Table has no array members.  Hold on a moment and I will upload a class that shows the bug right here.

**Comment by Raymond-Naseef** [comment]
Here is code causing the error:


```

import javassist.*;
import javassist.bytecode.*;
import javassist.bytecode.annotation.*;

@interface Entity {}

@interface Table { String[] textValues() default {}; }

public class Test {
    public static void main(String[] args) throws Exception {
        ClassPool classPool = ClassPool.getDefault();
        ClassFile cf = classPool.makeClass("TestSub").getClassFile();
        ConstPool constPool = cf.getConstPool();
        Annotation[] annotations = new Annotation[2];
        AnnotationsAttribute attrib =
                new AnnotationsAttribute(constPool, AnnotationsAttribute.visibleTag);
        Annotation annotation = new Annotation(constPool, classPool.get("Entity"));
        annotations[0] = annotation;
        // Add @Table(name="",schema="") to class
        annotation = new Annotation(constPool, classPool.get("Table"));
        annotation.addMemberValue("name", new StringMemberValue("name", constPool));
        annotation.addMemberValue("schema", new StringMemberValue("schema", constPool));
//        ArrayMemberValue blankMemberValueArray = new ArrayMemberValue(new AnnotationMemberValue(constPool), constPool);
//        blankMemberValueArray.setValue(new MemberValue[0]);
//        annotation.addMemberValue("textValues", blankMemberValueArray);
        annotations[1] = annotation;
        attrib.setAnnotations(annotations);
        cf.addAttribute(attrib);
    }
}
```

**Comment by chibash** [comment]
Still could not reproduce the bug.
Use the latest javassist.jar from the head.  I also checked your code in as src/test/Test.java.

**Comment by Raymond-Naseef** [comment]
I just downloaded your Test.java and ran Test in a new Eclipse project on OSX El Capitan with Java v1.8.0_05 seeing error every time.  Perhaps you would do me a favor and verify the version of Javassist in your classpath.  My project has only Test.java and the classpath and is a simple Java project.  If you want I can upload it to you (do not know a way to do that using github).

Good Luck!


Here is the classpath:

```
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" path="src/main/java"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
	<classpathentry kind="lib" path="/Users/raymondnaseef/.m2/repository/org/javassist/javassist/3.20.0-GA/javassist-3.20.0-GA.jar"/>
	<classpathentry kind="output" path="target/classes"/>
</classpath>

```

Here is console from running in Eclipse:

```
Exception in thread "main" java.lang.NullPointerException
	at javassist.bytecode.annotation.ArrayMemberValue.write(ArrayMemberValue.java:133)
	at javassist.bytecode.annotation.Annotation.write(Annotation.java:318)
	at javassist.bytecode.AnnotationsAttribute.setAnnotations(AnnotationsAttribute.java:248)
	at Test.main(Test.java:27)

```

Here is result running the code on my command line for the same code:

```
Raymonds-MBP-3:target raymondnaseef$ java -cp classes:/Users/raymondnaseef/.m2/repository/org/javassist/javassist/3.20.0-GA/javassist-3.20.0-GA.jar Test
Exception in thread "main" java.lang.NullPointerException
	at javassist.bytecode.annotation.ArrayMemberValue.write(ArrayMemberValue.java:133)
	at javassist.bytecode.annotation.Annotation.write(Annotation.java:318)
	at javassist.bytecode.AnnotationsAttribute.setAnnotations(AnnotationsAttribute.java:248)
	at Test.main(Test.java:27)

```

**Comment by chibash** [comment]
Can you download the latest javassist.jar from github.com and try again?

**Comment by Raymond-Naseef** [comment]
Great, I see the fix, thank you.  It seems my issue is partly due to using Maven.  I just reloaded library from Maven and got the old copy.  Should the corrected Test.java be available in Maven 3.20.0-GA or will it come out with the next release of Javassist?

**Comment by chibash** [comment]
The fix will be included in the next release.  I plan to release this month or next.
